src/ignore.c

```diff
@@ -370,13 +370,40 @@ int git_ignore__for_path(
 	return error;
 }
 
+char * concat_path_ign_file(git_buf *path)
+{
+	char *path_file = NULL;
+
+	path_file = git__malloc(path->size + 11);
+	if (path_file == NULL)
+		return NULL;
+
+	memcpy(path_file, path->ptr, path->size);
+	memcpy(path_file + path->size, GIT_IGNORE_FILE, 11);
+
+	return path_file;
+}
+
 int git_ignore__push_dir(git_ignores *ign, const char *dir)
 {
+	char *file_ign = NULL;
+	struct stat st;
+
 	if (git_buf_joinpath(&ign->dir, ign->dir.ptr, dir) < 0)
 		return -1;
 
 	ign->depth++;
 
+	// Do not add the .gitignore file if it not existing
+	file_ign = concat_path_ign_file(&ign->dir);
+	if (file_ign == NULL)
+		return -1;
+	if (p_stat(file_ign, &st) < 0) {
```

Comment:
I tested checking `errno` after `p_stat`:
```
    if (p_stat(file_ign, &st) < 0) {
      free(file_ign);
      if (errno != ENOENT && errno != ENOTDIR)
        return -1;
      else
        return 0;
    }
```
But the times went up a bit. At the moment caller of `git_ignore__push_dir` is not checking the return value.
Could that be a problem if `p_stat` fails with `(errno != ENOENT && errno != ENOTDIR)`?